tests/keras/backend/reference_operations.py

```diff
@@ -526,6 +528,101 @@ def resize_volumes(x, depth_factor, height_factor, width_factor, data_format):
     return x
 
 
+def ndim(x):
+    dims = x.shape
+    if dims is not None:
+        return len(dims)
+    return None
+
+
+def batch_dot(x, y, axes=None):
```

Comment:
Numpy has a `tensordot` function. I don't know much about it but can it help in our case?

Reply:
Indeed it is useful - if you look closely you will see I use it in the function. I tried to stick as closely to the tf backend implementation as possible to avoid unexpected bugs.